src/App.tsx

```diff
@@ -1,61 +1,132 @@
+import { Todo, TodoForRender } from './components/types/Todo';
 
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+import usersFromServer from './api/users';
+import todosFromServer from './api/todos';
+
+const initialData = (): TodoForRender[] => {
+  return (
+    todosFromServer.map((todo: Todo) => {
+      return ({
+        ...todo,
+        user: usersFromServer
+          .find((user: User) => user.id === todo.userId) || null,
+      });
+    })
+  );
+};
 
 export const App = () => {
+  const [todosToRender, setTodosToRender] = useState(initialData);
+  const [newTitle, setNewTitle] = useState('');
+  const [selectedUserId, setSelectedUserId] = useState('');
+  const [errTitle, setErrTitle] = useState(false);
+  const [errUser, setErrUser] = useState(false);
+
+  const todoUpdater = () => {
+    const maxId = todosToRender.reduce((max, { id }) => {
+      return (max < id) ? id : max;
+    }, 0);
+
+    const newTodo: TodoForRender = {
+      id: (maxId + 1),
+      title: newTitle,
+      completed: false,
+      userId: +selectedUserId,
+      user: usersFromServer
+        .find((user: User) => user.id === +selectedUserId) || null,
+    };
+
+    const updatedTodos = todosToRender.map(todo => ({ ...todo }));
+
+    updatedTodos.push(newTodo);
+
+    setTodosToRender(updatedTodos);
+    setNewTitle('');
+    setSelectedUserId('');
+  };
+
+  const handleSubmit = (event: React.FormEvent<HTMLFormElement>) => {
+    event.preventDefault();
+
+    if (!newTitle) {
+      setErrTitle(true);
+    }
+
+    if (!selectedUserId) {
+      setErrUser(true);
+    }
+
+    if (selectedUserId && newTitle) {
+      todoUpdater();
+    }
+  };
+
   return (
     <div className="App">
       <h1>Add todo form</h1>
 
-      <form action="/api/users" method="POST">
+      <form
+        action="/api/users"
+        method="POST"
+        onSubmit={handleSubmit}
+      >
         <div className="field">
-          <input type="text" data-cy="titleInput" />
-          <span className="error">Please enter a title</span>
+          <label>
+            <input
+              type="text"
+              data-cy="titleInput"
+              placeholder="Enter a title"
+              value={newTitle}
+              onChange={(event) => {
+                setNewTitle(event.target.value);
+                setErrTitle(false);
+              }}
+            />
+
+            {errTitle && (
+              <span className="error">Please enter a title</span>
+            )}
+          </label>
         </div>
 
         <div className="field">
-          <select data-cy="userSelect">
-            <option value="0" disabled>Choose a user</option>
-          </select>
+          <label>
+            <select
+              data-cy="userSelect"
+              value={selectedUserId}
+              onChange={(event) => {
+                setSelectedUserId(event.target.value);
+                setErrUser(false);
+              }}
```

Comment:
it is bad practice to create anonymous arrow functions in jsx
better to create function in component and pass reference here